src/Diagnostics.DataProviders/TokenService/TokenServiceBase.cs

```diff
@@ -97,12 +98,7 @@ public async Task StartTokenRefresh()
         /// </summary>
         public async Task<string> GetAuthorizationTokenAsync()
         {
-            if (!tokenAcquiredAtleastOnce)
-            {
-                var authResult = await acquireTokenTask;
-                return GetAuthTokenFromAuthenticationResult(authResult);
-            }
-
+            tokenAcquiredAtleastOnce.WaitOne();
```

Comment:
It seems here we are mixing sync code with Async code and this has a potential to deadlock

Reply:
Fixed